tests/integration/test_valid_data_in_request.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+# SPDX-License-Identifier: MIT
+
+import utils
+import time
+
+
+def test_valid_data_in_request(test_env):
+    """
+    Validate data in the request.
+
+    Process:
+    Send new request to the Cachito API
+    Send request to check status of existing request
+
+    Checks:
+    * Check that response code is 200
+    * Check in the response that state is complete
+    * Check that “packages” and “dependencies” keys have appropriate values
+    """
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    response_created_req = client.create_new_request(
+        payload={
+            "repo": test_env["get"]["repo"],
+            "ref": test_env["get"]["ref"],
+            "pkg_managers": test_env["get"]["pkg_managers"],
+        },
+    )
+
+    response = client.fetch_request(response_created_req.id)
+    assert response.status == 200
+
+    while True:
```

Comment:
Why not set the while loop condition to be `while response.data["state"] == "in_progress":`? Then you could remove your if branch below that breaks out of the loop.

Reply:
I'm not entirely sure what to do with this - @lcarva mentioned:
> Let's build some sort of timeout so this doesn't run forever if there's a bug in Cachito.

I like the approach with the suggested loop condition but is there a possibility that the state of request will stay in `in_progress` forever? In that case I could add a condition with timeout inside the loop. What do you think?

Reply:
You could do both:

```python
state = 'in_progress'
while state == 'in_progress':
  if timeout_expired:
    raise error
  response = client.fetch_request(response_created_req.id)
  state = response.data["state"]
```

It's a little awkward to initialize the state but I guess that's ok. Once we switch to python 3.8 we could use the [walrus operator](https://www.python.org/dev/peps/pep-0572/) and remove the state initialization.

I just want a timeout for now :)

Reply:
@ejegrova pointed out to me that a few lines above, we already fetch the response and have the status. So it should be possible to just use the response directly in the `while` expression: `while response.data["state"] == 'in_progress':`